hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/utils/ITTestHoodiePipeline.java

```diff
@@ -0,0 +1,220 @@
+import org.apache.flink.table.api.internal.TableEnvironmentImpl;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.runtime.typeutils.InternalTypeInfo;
+import org.apache.flink.table.types.logical.RowType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.hudi.sink.ITTestDataStreamWrite.EXPECTED;
+import static org.apache.hudi.utils.TestData.assertRowDataEquals;
+
+/**
+ *  IT test for test hoodie pipeline.
+ */
+public class ITTestHoodiePipeline {
+
+  public static final List<RowData> RESULT = new ArrayList<>();
+  @TempDir
+  File tempFile;
+  private StreamExecutionEnvironment execEnv;
+  private boolean isMor;
+  private Map<String, String> options = new HashMap<>();
+  private TableEnvironment streamTableEnv;
+
+  @BeforeEach
+  void beforEach() {
+    this.execEnv = StreamExecutionEnvironment.getExecutionEnvironment();
+    execEnv.getConfig().disableObjectReuse();
+    execEnv.setParallelism(4);
+    // set up checkpoint interval
+    execEnv.enableCheckpointing(4000, CheckpointingMode.EXACTLY_ONCE);
+    execEnv.getCheckpointConfig().setMaxConcurrentCheckpoints(1);
+
+    options.put(FlinkOptions.INDEX_TYPE.key(), "FLINK_STATE");
+    options.put(FlinkOptions.PATH.key(), tempFile.getAbsolutePath());
+    options.put(FlinkOptions.BUCKET_INDEX_NUM_BUCKETS.key(), "4");
+    options.put("table.type", HoodieTableType.MERGE_ON_READ.name());
+    options.put(FlinkOptions.INDEX_KEY_FIELD.key(), "id");
+    options.put(FlinkOptions.COMPACTION_DELTA_COMMITS.key(), "1");
+    options.put(FlinkOptions.TABLE_TYPE.key(), HoodieTableType.MERGE_ON_READ.name());
+    options.put(FlinkOptions.SOURCE_AVRO_SCHEMA_PATH.key(), Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("test_read_schema.avsc")).toString());
+
+    this.isMor = options.get(FlinkOptions.TABLE_TYPE.key()).equals(HoodieTableType.MERGE_ON_READ.name());
+
+    EnvironmentSettings settings = EnvironmentSettings.newInstance().build();
+    streamTableEnv = TableEnvironmentImpl.create(settings);
+    streamTableEnv.getConfig().getConfiguration().setInteger(ExecutionConfigOptions.TABLE_EXEC_RESOURCE_DEFAULT_PARALLELISM, 1);
+    Configuration execConf = streamTableEnv.getConfig().getConfiguration();
+    execConf.setString("execution.checkpointing.interval", "2s");
+    // configure not to retry after failure
+    execConf.setString("restart-strategy", "fixed-delay");
+    execConf.setString("restart-strategy.fixed-delay.attempts", "0");
+  }
+
+  @Test
+  public void testSink() throws Exception {
+    DataStream<RowData> dataStream = inputDataStream(this.execEnv, options, isMor);
+
```

Comment:
Maybe we can just move these tests into `ITTestDataStreamWrite`

Reply:
yes. I will move testSink() to ITTestDataStreamWrite and move testSource() to ITTestHoodieDataSource